backend/src/services/game.service.ts

```diff
@@ -97,3 +108,43 @@ export const allocatePlayerPunchlines = async (
 
   throw new ServiceError(ErrorType.playerId, "Player does not exist");
 };
+export const shuffleDiscardedSetups = async (game: Game): Promise<void> => {
+  if (game.setups.length <= 5) {
+    const discardedSetups: Setup[] = game.discardedSetups;
+    game.setups.push(...shuffle(discardedSetups));
```

Comment:
I think you should be doing a similar push to index 0 here as with shuffleDiscardPunchlines:

```
game.punchlines.push({
       $each: shuffledDiscard,
       $position: 0,
     });
```